controller/src/traffic_split.rs

```diff
@@ -166,17 +173,44 @@ async fn patch(
     client: kube::Client,
     params: &PatchParams,
     timeout: time::Duration,
-    FailoverUpdate { target, backends }: FailoverUpdate,
+    FailoverUpdate {
+        target,
+        backends,
+        primary_active,
+    }: FailoverUpdate,
 ) {
-    let namespace = target.namespace.expect("namespace must be set");
-    let api = Api::<TrafficSplit>::namespaced(client, &namespace);
-    let name = target.name;
+    let namespace = target.namespace.as_ref().expect("namespace must be set");
+    let api = Api::<TrafficSplit>::namespaced(client.clone(), namespace);
+    let name = &target.name;
     tracing::debug!("patching trafficsplit");
 
-    let patch = mk_patch(&name, &backends);
+    let event_reporter = events::Reporter {
+        controller: "linkerd-failover".to_string(),
```

Comment:
If we're using the `const FAILOVER` above maybe it makes sense to use a constant for this as well?